lib/galaxy/managers/quotas.py

```diff
@@ -151,10 +153,12 @@ def manage_users_and_groups_for_quota(self, quota, params, decode_id=None) -> st
                 raise ActionInputError("One or more invalid group id has been provided.")
             self.quota_agent.set_entity_quota_associations(quotas=[quota], users=in_users, groups=in_groups)
             self.sa_session.refresh(quota)
-            message = f"Quota '{quota.name}' has been updated with {len(in_users)} associated users and {len(in_groups)} associated groups."
-            return message
+            if len(quota.users) != len(in_users) or len(quota.groups) != len(in_groups):
+                return f"Quota '{quota.name}' has been updated with {len(in_users)} associated users and {len(in_groups)} associated groups."
```

Comment:
But the message would be misleading if one or the other is unchanged, right? (i.e., the if statement checks for a disjunction, but the message reports a conjunction)

Reply:
Hmm. How about the following (since a disjunction in the returned string also seems not useful)? 

```suggestion
            if len(quota.users) != len(in_users) and len(quota.groups) != len(in_groups):
                return f"Quota '{quota.name}' has been updated with {len(in_users)} associated users and {len(in_groups)} associated groups."
            elif len(quota.users) != len(in_users):
                return f"Quota '{quota.name}' has been updated with {len(in_users)} associated users."
            elif len(quota.groups) != len(in_groups):
                return f"Quota '{quota.name}' has been updated with {len(in_groups)} associated groups."
```